chainer/functions/activation/softmax.py

```diff
@@ -74,22 +74,38 @@ def backward(self, x, gy):
 
 
 def softmax(x, use_cudnn=True):
-    """Channelwise softmax function.
+    """Channel-wise softmax function.
 
     This function computes its softmax along the second axis. Let
-    :math:`x = (x_1, x_2, \\dots, x_d)^{\\top}` be the d dimensional index
-    array and :math:`f(x)` be the d dimensional input array. For each index
-    :math:`x` of the input array :math:`f(x)`, it computes the probability
-    :math:`p(x)` defined as
-    :math:`p(x) = {\\exp(f(x)) \\over \\sum_{x_2} \\exp(f(x))}`.
+    :math:`x=(x_1, x_2, \\dots, x_D)^{\\top}` be the D dimensional
+    input array.
+    For each input array :math:`x`, it computes the function :math:`f(x)`
+    defined as
+    :math:`f(x)={\\exp(x) \\over \\sum_{x_d} \\exp(x_d)}`.
 
     Args:
-        x (~chainer.Variable): Input variable.
+        X (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable.
+            A 2d (N, D) :math:`((s_1^1, ..., s_D^1), ..., (s_1^N, ..., s_D^N))`
```

Comment:
`softmax` can also be applied to more than 2-dimensional tensor, e.g., a minibatch of images shaped like `(N, C, H, W)`. In such case, the output will be `(N, C, H, W)`-shaped tensor and the summation over the channel axis will be always 1. In short, the output is not always "2d (N, D)", so I'd like you to fix this part ;)